docs/src/modules/components/AppTableOfContents.js

```diff
@@ -60,43 +60,6 @@ const useStyles = makeStyles((theme) => ({
   active: {},
 }));
 
-const renderer = new marked.Renderer();
```

Comment:
There were extra meassures taken to not re-create this on render and I don't know why. It's not like this is extra expensive. It's a simple class instantiation which sets some options. This was replaced by a module scope renderer and multiple global variables that were passed around. I don't think that indireaction was worth the marginal performance gain.

I'll run some web page tests just to be sure.

Reply:
Maybe the original assumption was that it's stateful, although on current inspection of the code it doesn't appear like it maintains any state, so should be 👍 to recreate.

Reply:
Being stateful would be an argument for re-creating it during render.

Reply:
I think that depends on what you're trying to achieve, e.g. if the renderer's lifecycle follows the component's lifecycle. But if you want every single render to start from a clean slate, then yes, state would be an argument for recreating.

Reply:
Oh yeah. Memoization could be a form of statefulness that is definitely safe to use during render.

Reply:
No noticeable difference: https://www.webpagetest.org/video/compare.php?tests=200414_2E_edd5a49d96045d92f7f2314f6ece34d3,200414_YY_6fc7ac66001198a18d74a0be13e0dc5c